s3fs/core.py

```diff
@@ -1153,7 +1153,7 @@ async def _put_file(
         callback.set_size(size)
 
         if "ContentType" not in kwargs:
-            content_type, _ = mimetypes.guess_type(lpath)
+            content_type, _ = mimetypes.guess_type(rpath)
```

Comment:
This would preserve current behaviour if indeed there are cases where the remote path is extensionless.
```suggestion
            content_type = mimetypes.guess_type(rpath)[0] or mimetypes.guess_type(lpath)[0]
```